Gems/Terrain/Code/Source/TerrainRenderer/ClipmapBounds.cpp

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <TerrainRenderer/ClipmapBounds.h>
+#include <AzCore/Casting/numeric_cast.h>
+#include <AzCore/Math/MathUtils.h>
+
+namespace Terrain
+{
+    bool ClipmapBoundsRegion::operator==(const ClipmapBoundsRegion& other) const
+    {
+        return m_localAabb == other.m_localAabb && m_worldAabb.IsClose(other.m_worldAabb);
+    }
+    
+    bool ClipmapBoundsRegion::operator!=(const ClipmapBoundsRegion& other) const
+    {
+        return !(*this == other);
+    }
+
+    ClipmapBounds::ClipmapBounds(const ClipmapBoundsDescritor& desc)
+        : m_size(desc.m_size)
+        , m_halfSize(desc.m_size >> 1)
+        , m_margin(AZ::GetMax<uint32_t>(desc.m_margin, 1))
+        , m_scale(desc.m_scale)
+        , m_rcpScale(1.0f / desc.m_scale)
+    {
+        AZ_Assert(m_size % m_margin == 0, "ClipmapBounds: m_size should be a mulitple of m_margin")
```

Comment:
Chatted offline, while a margin of 4 with 100 would work fine because 100 % 4 == 4, I don't think this assert is actually necessary so I'm removing it.